### PR TITLE
Enable Console lib split

### PR DESCRIPTION
**Pull request by markstory**
Key changes here:

* The `ConsoleErrorHandler` has moved to the Error package. This lets us have much simpler dependencies in the Console package.
* The Command base class includes a bunch of integrations with the rest of CakePHP that make it burdensome to use. By adding a CommandInterface we can narrow the dependencies for standalone usage.

Applications that use the console standalone library would need to setup their own `run()` method that uses the optionparser if they want. I'm open to adding a leaner Command base class but don't have a good name for it right now. Any suggestions are welcome.

### CODE REVIEW

**Comment by ADmad** [comment]
> I'm open to adding a leaner Command base class but don't have a good name for it right now.

We have `BasePlugin`,  `BaseApplication` etc. so perhaps `BaseCommand`?

**Comment by saeideng** [comment]
> I'm open to adding a leaner Command base class but don't have a good name for it right now. Any suggestions are welcome.

for new class or for renaming `src/console/Command` to new name ?
if is new  what you want to put into it ?

**Comment by markstory** [comment]
Renaming/changing the existing `Command` class would cause a big break for applications and I don't think it is necessary. A name like `BaseCommand` or `AbstractCommand` could work.

**Comment by markstory** [comment]
I've added a `BaseCommand` class. I'm happy to rename it if folks have an alternative name.

File: src/Console/BaseCommand.php [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    }
+
+    /**
+     * Implement this method with your command's logic.
+     *
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return int|null The exit code or null for success
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
```
Comment (saeideng) [inline src/Console/BaseCommand.php]: since we have `command` class that extended this class 
probably you can define  this method as abstract

Comment (markstory) [reply src/Console/BaseCommand.php]: Good idea :+1:

**Comment by markstory** [comment]
Anything else for this?

File: src/Console/BaseCommand.php [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+     * the options and arguments are validated and processed.
+     *
+     * @return void
+     */
+    public function initialize(): void
+    {
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (ADmad) [inline src/Console/BaseCommand.php]:
```suggestion
     * @inheritDoc
```

**Pull request merged**